### DISCUSSION TITLE
[Question]: Undenstanding: RAG_USE_FULL_CONTEXT

### DISCUSSION THREAD

**Discussion by ala85** [danny-avila/LibreChat#4260] [Q&A]
### What is your question?

Hi, according to https://github.com/danny-avila/LibreChat/pull/2169, 

> RAG_USE_FULL_CONTEXT will retrieve the full document and add it to the prompt instead of using semantic retrieval.

I am trying to understand what it means.

Currently, the chat is able to answer with a certain context, if you select the files you are interested with (and that a user has previously uploaded). My interpretation of the parameter was that if enabled, ALL documents a user has uploaded would be automatically used as context, without the need for the user to drag and drop it in the chat form. Is this correct? Because after a few trials does not seem so and wanted to be confirmed.

### More Details

Already provided.

### What is the main subject of your question?

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by edmcquinn** [comment]
@ala85 No this is related to the RAG implementation. Normally when you upload a file to librechat it tries to find similar content in your document via RAG this only includes some of the information that most closely matches your question. All this variable is doing is sending the entire context of the file instead of using rag I believe.
